react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+export default class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { selected: 0 };
+    this.handleRight = this.handleRight.bind(this);
+    this.handleLeft = this.handleLeft.bind(this);
+    this.handleDots = this.handleDots.bind(this);
+    this.length = this.props.gallery.length;
+    this.timer = setInterval(() => {
+      if (this.state.selected === (this.length - 1)) {
+        this.setState({ selected: 0 });
+      } else {
+        this.setState({ selected: this.state.selected + 1 });
+      }
+    }, 5000);
```

Comment:
You should do this in `componentDidMount` instead of the constructor, in case the `render()` method throws an error for some reason. Otherwise, you'll create a memory leak.

Reply:
https://reactjs.org/docs/react-component.html#componentdidmount